projects/packages/connection/src/class-webhooks.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Connection Webhooks class.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection;
+
+use Automattic\Jetpack\Roles;
+use Automattic\Jetpack\Tracking;
+use Jetpack_Options;
+
+/**
+ * Connection Webhooks class.
+ */
+class Webhooks {
+
+	/**
+	 * The Connection Manager object.
+	 *
+	 * @var Manager
+	 */
+	private $connection;
+
+	/**
+	 * Webhooks constructor.
+	 *
+	 * @param Manager $connection The Connection Manager object.
+	 */
+	public function __construct( $connection ) {
+		$this->connection = $connection;
+	}
+
+	/**
+	 * Initialize the webhooks.
+	 *
+	 * @param Manager $connection The Connection Manager object.
+	 */
+	public static function init( $connection ) {
+		$webhooks = new static( $connection );
+
+		add_action( 'init', array( $webhooks, 'controller' ) );
+	}
+
+	/**
+	 * The "controller" decides which handler we need to run.
+	 */
+	public function controller() {
+		// The nonce is verified in specific handlers.
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( empty( $_GET['handler'] ) || empty( $_GET['action'] ) || 'jetpack-connection-webhooks' !== $_GET['handler'] ) {
+			return;
+		}
+
+		// The nonce is verified in specific handlers.
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		switch ( $_GET['action'] ) {
+			case 'authorize':
+				$this->handle_authorize();
+				break;
+		}
+
+		exit;
+	}
+
+	/**
+	 * Perform the authorization action.
+	 */
+	public function handle_authorize() {
+		if ( $this->connection->is_active() && $this->connection->is_user_connected() ) {
+			$redirect_url = apply_filters( 'jetpack_client_authorize_already_authorized_url', admin_url() );
+			wp_safe_redirect( $redirect_url );
+
+			return;
+		}
+		do_action( 'jetpack_client_authore_processing' );
```

Comment:
I think this is a typo. It should probably be:

```suggestion
		do_action( 'jetpack_client_authorize_processing' );
```

Reply:
Good catch, thanks Kim!
I added that to unit tests, as I should've done earlier.